### PR TITLE
Start documenting release process

### PR DESCRIPTION
**Pull request by nathanleclaire**
cc @docker/machine-maintainers 

A summary / checklist of what I've been doing for the releases, to be followed up soon by a few clarifications / automations to our current `Makefile` and scripts.

Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+1.  **Version Bump** -- When all commits for the release have been merged into
+master and/or the release branch, submit a pull request bumping the `Version`
+variable in `version/version.go` to the release version.  Merge said pull
+request.  
+2. **Compile Binaries** -- Pull down the latest changes to master and
+cross-compile the core binary and plugin binaries for each supported OS /
+architecture combination.  As you may notice, this can potentially take a _very_
+long time so you may want to use a script like [this one for cross-compiling
+them on DigitalOcean in
+parallel](https://gist.github.com/nathanleclaire/7f62fc5aa3df19a50f4e).
```
Comment (hairyhenderson) [inline docs/RELEASE.md]: Nice script - any reason why this shouldn't be in the `docker/machine` repo?

Comment (hairyhenderson) [inline docs/RELEASE.md]: Er, never mind - I see now from the original description that you're planning on doing some follow-ups :grin:

**Comment by dgageot** [comment]
@nathanleclaire The text looks good to me.
Please fix the command that prints the author names. Some names are truncated. You could use `git log v0.5.0.. --format="%aN" --reverse | sort -u | awk '{printf "- %s\n", $0 }'`

**Comment by nathanleclaire** [comment]
Nice call @dgageot, I've updated.  And sorry @jeanlaurent, I updated your name in the release notes ;D

**Comment by nathanleclaire** [comment]
@SvenDowideit Let me know what to change to get this passing the docs build.  It just says "No test results found" :)

**Comment by SvenDowideit** [comment]
@nathanleclaire thank you for being great shakeout testers :) (aka sorry)

no tests is a bizzarly hard coded string in the jenkins plugin we use - click the link and it will tell you what's up.

but it should now be fixed - you can tell jenkins to 'retry' the job from there too.

**Comment by SvenDowideit** [comment]
@nathanleclaire  oh, nice - now the validation is correct :) 

this time, its because your PR needs a rebase so that it gets the fixes from #2329

**Comment by nathanleclaire** [comment]
OK, rebased.

**Comment by SvenDowideit** [comment]
your new RELEASE.md file in the `docs` dir needs the hugo frontmatter - i presume you don't want this published, so really all it needs is the `draft=true` that's in the other md files.

**Comment by nathanleclaire** [comment]
Added the front matter.

**Comment by dgageot** [comment]
LGTM

**Pull request merged**